common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -288,79 +314,50 @@ def transcript(self, request, dispatch):
-                response = self.get_static_transcript(request, transcripts)
-                if response.status_code == 404 and feature_enabled:
-                    # Try to get transcript from edx-val as a last resort.
-                    transcript = get_video_transcript_content(self.edx_video_id, self.transcript_language)
-                    if transcript:
-                        transcript_conversion_props = dict(transcript, output_format=Transcript.SJSON)
-                        transcript = convert_video_transcript(**transcript_conversion_props)
-                        response = Response(
-                            transcript['content'],
-                            headerlist=[('Content-Language', self.transcript_language)],
-                            charset='utf8',
-                        )
-                        response.content_type = Transcript.mime_types[Transcript.SJSON]
-
-                return response
-            except (UnicodeDecodeError, TranscriptsGenerationException) as ex:
-                log.info(six.text_type(ex))
-                response = Response(status=404)
-            else:
-                response = Response(transcript, headerlist=[('Content-Language', language)])
-                response.content_type = Transcript.mime_types['sjson']
+                if is_bumper:
+                    content, filename, mimetype = get_transcript_from_contentstore(
+                        self,
+                        self.transcript_language,
+                        Transcript.SJSON,
+                        transcripts
+                    )
+
+                    return self.make_transcript_http_response(
+                        content,
+                        filename,
+                        self.transcript_language,
+                        mimetype
+                    )
 
+                content, filename, mimetype = get_transcript(
+                    self,
+                    lang=self.transcript_language,
+                    output_format=Transcript.SJSON,
+                    youtube_id=request.GET.get('videoId', None),
+                )
+                response = self.make_transcript_http_response(
+                    content,
+                    filename,
+                    self.transcript_language,
+                    mimetype
+                )
+            except NotFoundError as ex:
+                log.info(six.text_type(ex))
+                return self.get_static_transcript(request, transcripts)
         elif dispatch == 'download':
             lang = request.GET.get('lang', None)
+
             try:
-                transcript_content, transcript_filename, transcript_mime_type = self.get_transcript(
-                    transcripts, transcript_format=self.transcript_download_format, lang=lang
-                )
-            except (KeyError, UnicodeDecodeError):
+                content, filename, mimetype = get_transcript(self, lang)
```

Comment:
we can construct response here, to avoid multiple return statements:
```bash
response = self.make_transcript_http_response(
                content,
                filename,
                self.transcript_language,
                mimetype
            )
```